### PR TITLE
Feat: delete the sync deployment after deploy complete

### LINKED ISSUE #70: Delete the sync Deployment after deploy complete.
We start a deployment called `curve-sync-config` as a server to synchronize the config files of various services. But this service is not needed after the deployment is completed, so this deployment needs to be deleted, otherwise unnecessary resources will be wasted.

我们为了同步所需要的配置文件，在部署的时候需要启动一个叫`curve-sync-config`的deployment，现在我们需要在确定同步成功之后，或者在整个集群部署完成之后，删除这个deployment,因为这个deployment已经无用并且会占用系统的一些资源。

### PR DESCRIPTION
**Pull request by lizonglingo**
This pr resolves #70 .

Closes #70 .

### CODE REVIEW

File: pkg/controllers/client.go [hunk 1]
```diff
@@ -154,3 +170,161 @@ func getReadConfigJobLabel(c *daemon.Cluster) map[string]string {
...
+
+	time.Sleep(1 * time.Minute)
+
+	if c.Kind == config.KIND_CURVEBS {
+		logger.Debugf("node count is %d, wanted chunk server count is %d", nodeCount, wantChunkServer)
+	} else if c.Kind == config.KIND_CURVEFS {
+		logger.Debugf("node count is %d", nodeCount)
+	}
+
+	checkTicker := time.NewTicker(30 * time.Second)
```
Comment (caoxianfei1) [inline pkg/controllers/client.go:205]: The for{} never stop if something wrong, you can add a timeout time

File: pkg/controllers/client.go [hunk 2]
```diff
@@ -154,3 +170,161 @@ func getReadConfigJobLabel(c *daemon.Cluster) map[string]string {
...
+
+	err := c.Context.Clientset.AppsV1().Deployments(c.Namespace).Delete(syncConfigDeployment, &metav1.DeleteOptions{})
+	if err != nil {
+		logger.Errorf("failed to delete deployment about \"curve-sync-config\", error: %s", err)
+	}
+
+	logger.Infof("cluster is deployed, deployment about \"curve-sync-config\" will be deleted")
+}
+
+func isAllReplicasReady(deployment apps.Deployment) bool {
```
Comment (caoxianfei1) [inline pkg/controllers/client.go:318]: The deployment operation and job operation code put the k8sutil dir is better.

File: pkg/controllers/client.go [hunk 3]
```diff
@@ -86,6 +98,10 @@ func createSyncDeployment(c *daemon.Cluster) error {
 			return err
 		}
 	}
+
+	// delete the SyncConfigDeployment after the cluster is deployed.
```
Comment (caoxianfei1) [inline pkg/controllers/client.go:102]: using channel to get delete result (true or false) and log the reason if delete the deployment failed.

File: pkg/controllers/client.go [hunk 4]
```diff
@@ -154,3 +170,161 @@ func getReadConfigJobLabel(c *daemon.Cluster) map[string]string {
...
+
+func isAllReplicasReady(deployment apps.Deployment) bool {
+	if deployment.Status.Replicas == deployment.Status.ReadyReplicas {
+		return true
+	}
+	return false
+}
+
+func isJobCompleted(job batch.Job) bool {
+	if *job.Spec.Completions == job.Status.Succeeded {
```
Comment (caoxianfei1) [inline pkg/controllers/client.go:326]: the job status is failed or other status sometimes.

**Comment by lizonglingo** [comment]
ok, I'll make some changes and additions @caoxianfei1

**Comment by caoxianfei1** [comment]
> ok, I'll make some changes and additions @caoxianfei1

ok

**Comment by lizonglingo** [comment]
new code has been submitted @caoxianfei1

File: pkg/controllers/client.go [hunk 5]
```diff
@@ -154,3 +157,22 @@ func getReadConfigJobLabel(c *daemon.Cluster) map[string]string {
 	labels["curve"] = c.Kind
 	return labels
 }
+
+// deleteSyncConfigDeployment delete the SyncConfigDeployment after the cluster is deployed.
```
Comment (caoxianfei1) [inline pkg/controllers/client.go:161]: I think that put the func in k8sutil fold and deployment file is better

Comment (lizonglingo) [reply pkg/controllers/client.go]: ok, I've adjusted its directory

**Review by caoxianfei1 — APPROVED** [review]

**Comment by caoxianfei1** [comment]
lgtm

**Pull request merged**